evaluate/task/task-transpile.go

```diff
@@ -170,49 +162,69 @@ func (t *TaskTranspile) Run(ctx evaltask.Context) (repositoryAssessment map[eval
+			return nil, "", pkgerrors.Errorf("the language extension %q is not supported", filepath.Ext(file.Name()))
+		}
+		originFilePathsWithLanguage[filepath.Join(packagePathAbsolute, "implementation", file.Name())] = originLanguage
 	}
-	originFilePath = filepath.Join("implementation", files[0])
 
-	stubFilePath, err = packageSourceFile(fileLogger, packagePathAbsolute, ctx.Language)
+	stubFilePath, err = packageSourceFile(logger, packagePathAbsolute, ctx.Language)
 	if err != nil {
-		return "", "", err
+		return nil, "", err
 	}
 
-	return originFilePath, stubFilePath, nil
+	return originFilePathsWithLanguage, stubFilePath, nil
 }
 
 // validateTranspileRepository checks if the repository for the "transpile" task is well-formed.
 func validateTranspileRepository(logger *log.Logger, repositoryPath string, destinationLanguage language.Language) (err error) {
 	logger.Printf("validating repository %q", repositoryPath)
 
-	var originLanguage language.Language
-	if _, ok := destinationLanguage.(*golang.Language); ok {
-		originLanguage = &java.Language{}
-	} else {
-		originLanguage = &golang.Language{}
-	}
-
 	packagePaths, err := repositoryOnlyHasPackages(repositoryPath)
 	if err != nil {
 		return err
 	}
 
 	for _, packagePath := range packagePaths {
-		// Validate the implementation folder.
-		files, err := originLanguage.Files(logger, filepath.Join(packagePath, "implementation"))
+		implementationDirectoryPath := filepath.Join(packagePath, "implementation")
+		files, err := os.ReadDir(implementationDirectoryPath)
 		if err != nil {
 			return pkgerrors.WithStack(err)
-		} else if len(files) != 1 {
-			return pkgerrors.Errorf("package %q must have an \"implementation\" directory with just one %s source file to transpile", packagePath, originLanguage.Name())
-		} else if strings.HasSuffix(files[0], originLanguage.DefaultTestFileSuffix()) {
-			return pkgerrors.Errorf("package %q must have an \"implementation\" directory with only a %s source file, but found a test file %q", packagePath, originLanguage.Name(), files[0])
+		}
+
+		// Check if the implementation directory is well-formed.
+		for _, file := range files {
+			if file.IsDir() {
+				return pkgerrors.Errorf("the implementation directory %q must contain only source code files to transpile, but found one directory: %q", implementationDirectoryPath, file.Name())
+			}
+
+			originLanguage, ok := language.LanguagesByFileExtension[filepath.Ext(file.Name())]
+			if !ok {
+				return pkgerrors.Errorf("the language extension %q is not supported", filepath.Ext(file.Name()))
+			}
+			if strings.HasSuffix(file.Name(), originLanguage.DefaultTestFileSuffix()) {
+				return pkgerrors.Errorf("the implementation directory %q must contain source files, but found a test file %q", implementationDirectoryPath, files[0])
+			}
+			// Check if the implementation directory only contains one source file per language.
+			originLanguageFiles, err := originLanguage.Files(logger, implementationDirectoryPath)
```

Comment:
Nitpick: rather than getting all other files every time, could also just add a `map[language.Language]bool` and throw an error in case we encounter a file where the language is already in the map.